src/equinoxe/equinoxe.ml

```diff
@@ -591,14 +594,13 @@ module Make (B : Backend) : API with type 'a io = 'a B.io = struct
     let build ?hostname ?tags ~plan ~os ~location () =
       { hostname; tags; plan; os; location }
 
-    let opt_to_list :
-          'a.
-          string ->
-          ('a -> Ezjsonm.value) ->
-          'a option ->
-          (string * Ezjsonm.value) list =
-     fun name fn v ->
-      match v with Some value -> [ (name, fn value) ] | None -> []
+    let opt_to_list name fn = function
+      | Some value -> [ (name, fn value) ]
+      | None -> []
+
+    let custom_to_list = function
+      | Custom_ipxe url -> [ ("ipxe_script_url", `String url) ]
```

Comment:
Very cool! Perhaps it could be ``[`URL of Uri.t | `Script of string]`` as apparently it's possible to pass the ipxe script directly via `"userdata"`? https://developer.equinix.com/catalog/metalv1#operation/createDevice https://metal.equinix.com/developers/docs/operating-systems/custom-ipxe/#provisioning-with-custom-ipxe